src/modules/videoconference/VideoConferenceModule/Toolbar.cpp

```diff
@@ -154,18 +154,27 @@ LRESULT Toolbar::WindowProcessMessages(HWND hwnd, UINT msg, WPARAM wparam, LPARA
         static bool previousShow = false;
         bool show = false;
 
-        if (toolbar->cameraInUse)
+        if (toolbar->ToolbarHide == L"Never")
         {
-            show = toolbar->HideToolbarWhenUnmuted ? toolbar->microphoneMuted || toolbar->cameraMuted : true;
+            show = true;
         }
-        else if (toolbar->previouscameraInUse)
+        else if (toolbar->ToolbarHide == L"When both camera and microphone are muted")
         {
-            VideoConferenceModule::unmuteAll();
+            if(!toolbar->previouscameraInUse && toolbar->cameraInUse && !toolbar->moduleSettingsUpdateScheduled)
+            {
+                VideoConferenceModule::muteAll();
+            }
+            show = !(toolbar->microphoneMuted && (toolbar->cameraMuted || !toolbar->cameraInUse));
         }
-        else
+        else if (toolbar->ToolbarHide == L"When both camera and microphone are unmuted")
```

Comment:
let's create cosntants for these string values, because it's easy to make a typo o/w.

Reply:
![Screenshot_20221006-152442.png](https://user-images.githubusercontent.com/65828559/194324902-213f5463-0037-4855-8589-be016f1cad35.png)